chain/arweave/src/chain.rs

```diff
@@ -107,12 +110,11 @@ impl Blockchain for Chain {
         unimplemented!("This chain does not support Dynamic Data Sources. is_refetch_block_required always returns false, this shouldn't be called.")
     }
 
-    async fn new_firehose_block_stream(
+    async fn new_block_stream(
         &self,
         deployment: DeploymentLocator,
-        block_cursor: FirehoseCursor,
+        store: Arc<dyn WritableStore>,
```

Comment:
I get that I thought of adding an enum where you could pass one or the other but leads to some iffy code. I don't like this pattern either but it sort of hides the difference between firehose and rpc. Alternatively I could move the 2 functions needed to a separate trait and require that here. This way it's a lot clearer what we need from this argument. What do you think?

Reply:
Can't you just have the caller call `store.block_cursor()` and `store.block_ptr()` and pass those in? All I am after is to make it clear that the block stream has no real interest in the `WritableStore`

Reply:
I completely agree I just wanted to avoid the caller having to deal with the firehose/non firehose duality since they don't really need to know anything about it. By providing the store (which I agree is a lot of surface for what we actually need) we can let the callee decide which one to use. ie ignore block_cursor if not needed. 

I absolutely agree that this doesn't seem good as is though so I'll try to find a better solution.